tests/logging_db/mod.rs

```diff
@@ -0,0 +1,164 @@
+#[test]
```

Comment:
Other tests I would like to see:
- Coverage of all the builtin types (e.g. scalars, str, never, etc.)
- Coverage of functions (fn ptr and fn def)

It might be useful to "wrap" the main tests locally just to have a broader test set, to see if we missed anything.

Reply:
Welp I just scrolled up and saw these are coveraged. Well done :)

Reply:
> It might be useful to "wrap" the main tests locally just to have a broader test set, to see if we missed anything.

We tried to do this at first, but found difficulty hooking into the existing test function which used `ChalkDatabase` to run everything - we talked a bit about this on zulip. It'd be possible to write this in `solve_goal`, and use some of the suggestions (like maybe putting a `logging` flag in `ChalkDatabase`). However, I thought that it would lead to diminishing returns compared to tests directed at `LoggingIrDatabase`, and the huge amount of extra test harness code wouldn't be worth it.

My biggest objection was that running it on the existing test suite wouldn't hit too many extra edge cases, since we'd be running it on tests which weren't designed to test `LoggingIrDatabase` edge cases. If the cost is too great, like changing things in `ChalkDatabase`, then it'd be a bust.

But thinking about this again, there might be a reasonable strategy. We could just run our test harness at the end of `solve_goal`, rather than trying to integrate our tests into `solve_goal`. To run this on all tests, we would need to essentially fully duplicate `solve_goal` in our test code. But our current code still partially duplicates it, and the only reason it isn't full duplication is to stay minimal.

If it's OK that we fully duplicate `solve_goal`'s functionality, then we'll do that.

Reply:
Nah, don't worry about it.